utils/ccdatofhir/mapping_resource_immunization.bal

```diff
@@ -0,0 +1,262 @@
+// specific language governing permissions and limitations
+// under the License.
+
+// --------------------------------------------------------------------------------------------#
+// Source C-CDA to FHIR - Resource Mappings
+// --------------------------------------------------------------------------------------------#
+
+import ballerina/log;
+import ballerinax/health.fhir.r4;
+
+# Map CCDA Immunization Activity to FHIR Immunization.
+#
+# + substanceAdministrationElement - CCDA Immunization Activity Element
+# + return - FHIR Immunization Resource
+public isolated function mapCcdaImmunizationToFhir(xml substanceAdministrationElement) returns r4:Immunization? {
+    if isXMLElementNotNull(substanceAdministrationElement) {
+        r4:Immunization immunization = {patient: {}, occurrenceDateTime: "", occurrenceString: "", vaccineCode: {}, status: "not-done"};
+
+        xml idElement = substanceAdministrationElement/<v3:id|id>;
+        xml statusCodeElement = substanceAdministrationElement/<v3:statusCode|statusCode>;
+        xml effectiveTimeElement = substanceAdministrationElement/<v3:effectiveTime|effectiveTime>;
+        xml repeatNumberElement = substanceAdministrationElement/<v3:repeatNumber|repeatNumber>;
+        xml routeCodeElement = substanceAdministrationElement/<v3:routeCode|routeCode>;
+        xml approachSiteCodeElement = substanceAdministrationElement/<v3:approachSiteCode|approachSiteCode>;
+        xml doseQuantityElement = substanceAdministrationElement/<v3:doseQuantity|doseQuantity>;
+        xml vaccineCodeElement = substanceAdministrationElement/<v3:consumable|consumable>/<v3:manufacturedProduct|manufacturedProduct>/<v3:manufacturedMaterial|manufacturedMaterial>/<v3:code|code>;
+        xml lotNumberTextElement = substanceAdministrationElement/<v3:consumable|consumable>/<v3:manufacturedProduct|manufacturedProduct>/<v3:manufacturedMaterial|manufacturedMaterial>/<v3:lotNumberText|lotNumberText>;
+        xml performerElement = substanceAdministrationElement/<v3:performer|performer>;
+        xml entryRelationship = substanceAdministrationElement/<v3:entryRelationship|entryRelationship>;
+
+        int index = 0;
+        foreach xml idElem in idElement {
+            r4:Identifier? mapCcdaIdToFhirIdentifierResult = mapCcdaIdToFhirIdentifier(idElem);
+            if mapCcdaIdToFhirIdentifierResult is r4:Identifier {
+                immunization.identifier[index] = mapCcdaIdToFhirIdentifierResult;
+                index = index + 1;
+            }
+        }
+
+        immunization.status = mapCcdaStatusCodeToFhirImmunizationStatus(statusCodeElement);
+
+        r4:dateTime? mapCCDAEffectiveTimetoFHIRDateTimeResult = mapCcdaDateTimeToFhirDateTime(effectiveTimeElement);
+        if mapCCDAEffectiveTimetoFHIRDateTimeResult is r4:dateTime {
+            immunization.occurrenceDateTime = mapCCDAEffectiveTimetoFHIRDateTimeResult;
+        }
+
+        string|error? repeatNumber = repeatNumberElement.value;
```

Comment:
```suggestion
        string|error repeatNumber = repeatNumberElement.value;
```
I am seeing unnecessary usages of `?`.  `?` is used only if the expression from the right hand side returns nil.

Reply:
Please refer https://discord.com/channels/957996897782616114/1123904724425523210/1124169449356931072